src/app/tasks/tasks.component.ts

```diff
@@ -1,67 +1,204 @@
-// import {selectBoardTasks} from '../store/boards.selectors'
-import { Observable } from 'rxjs';
-import { addTaskOpen, editTaskOpen, modalOpen } from '../store/modal.action';
+
 import {
-  addTaskModalSelector,
-  editTaskSelector,
-  showBoardModalSelector,
-} from '../store/modal.selectors';
-import { AddTaskFormComponent } from '../forms/addTask/addTask.component';
+  GetTasks,
+  deleteTask,
+  updateTask,
+} from '../store/actions/tasks.action';
+
+import { Observable, skipWhile, tap } from 'rxjs';
 import { ActivatedRoute } from '@angular/router';
+import { selectAllTasks } from '../store/selectors/tasks.selectors';
+
+import {
+  faArrowDownShortWide,
+  faArrowUpWideShort,
+  faBrush,
+  faPenToSquare,
+  faPlus,
+  faTrashCan,
+} from '@fortawesome/free-solid-svg-icons';
+import {
+  CdkDragDrop,
+  moveItemInArray,
+  transferArrayItem,
+} from '@angular/cdk/drag-drop';
+import { deleteComment } from '../store/actions/comment.actions';
+import { updateColor } from '../store/actions/board.action';
+import { Board } from '../models/board.model';
+import { selectBoard, selectUI } from '../store/selectors/boards.selectors';
+import { getBoardById } from '../store/actions/board.action';
 
 @Component({
   selector: 'tasks-app',
   templateUrl: './tasks.component.html',
   styleUrls: ['./tasks.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class TasksComponent implements OnInit {
-  tasks$: Observable<TaskI[]> = this.store.select((state: any) => state.tasks);
-  addTaskModal: Observable<boolean> = this.store.select(addTaskModalSelector);
-  currentBoard: any;
+  uiPreferences$: any = this.store.select(selectUI);
+  tasks$: Observable<Task[]> = this.store.select(selectAllTasks);
+  board$: Observable<Partial<Board>> = this.store.select(selectBoard);
+  addModal: boolean = false;
+  editModal: boolean = false;
+  addCommentModal: boolean = false;
+
+  showComments = {
+    taskId: '',
+    show: false,
+  };
+
+  currentBoard!: any;
   statusOutput!: string;
   sortOrder: string = '';
   filter: string = '';
   compositeFilter: string = '';
-  // activeTask: string;
-  activeTask!: TaskI;
-  editModal: Observable<boolean> = this.store.select(editTaskSelector);
+  activeTask!: Task;
+
+  columnBcg = {
+    col1: '#c4c2c2',
+    col2: '#c4c2c2',
+    col3: '#c4c2c2',
+  };
+
+  showColorpicker = {
+    col1: false,
+    col2: false,
+    col3: false,
+  };
+
+  icons = {
+    desc: faArrowDownShortWide,
+    asc: faArrowUpWideShort,
+    delete: faTrashCan,
+    edit: faPenToSquare,
+    add: faPlus,
+    paint: faBrush,
+  };
 
   ngOnInit() {
     this.currentBoard = this.activatedRoute.snapshot.paramMap.get('id');
+    this.store.dispatch(getBoardById({ id: this.currentBoard }));
     this.store.dispatch(GetTasks({ id: this.currentBoard }));
+
+    this.uiPreferences$
+      .pipe(
+        skipWhile((res) => !res),
+        tap((res: any) => {

```

Comment:
you can simplify - instead of res - {col1, col2, col3} and in columnBg delete res..